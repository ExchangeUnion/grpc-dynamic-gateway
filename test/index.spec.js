/* global describe, it, expect */
const { generateSwagger, convertParams, convertUrl, convertBody, getParamsList } = require('..')

describe('gRPC Dynamic Gateway', () => {
  describe('generateSwagger()', () => {
    it('should generate correct swagger', () => {
      expect(generateSwagger([`${__dirname}/../example/api.proto`, `${__dirname}/test.proto`])).toMatchSnapshot()
    })
  })

  describe('convertParams()', () => {
    it('should handle /v1/hi/{name}', () => {
      const req = {
        body: {
          v1: true,
          v2: false
        },
        params: {
          name: 'Cool'
        }
      }
      expect(convertParams(req, '/v1/hi/{name}')).toMatchSnapshot()
    })
  })

  describe('convertUrl()', () => {
    it('should correctly convert /v1/hi/{name} into express URL', () => {
      expect(convertUrl('/v1/hi/{name}')).toMatchSnapshot()
    })
    it('should correctly convert /{version}/hi/{name}/{cool} into express URL', () => {
      expect(convertUrl('/{version}/hi/{name}/{cool}')).toMatchSnapshot()
    })
  })

  describe('convertBody()', () => {
    it('should handle {cool: true}, *', () => {
      expect(convertBody({cool: true}, '*')).toMatchSnapshot()
    })
    it('should handle {cool: true}, cool', () => {
      expect(convertBody({cool: true}, 'cool')).toMatchSnapshot()
    })
  })

  describe('getParamsList()', () => {
    it('should find params in /v1/hi/{name}', () => {
      expect(getParamsList('/v1/hi/{name}')).toMatchSnapshot()
    })
    it('should find params in /{version}/hi/{name}/{cool}', () => {
      expect(getParamsList('/{version}/hi/{name}/{cool}')).toMatchSnapshot()
    })
  })
})
