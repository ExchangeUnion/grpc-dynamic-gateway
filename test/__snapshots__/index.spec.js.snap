// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gRPC Dynamic Gateway convertBody() should handle {cool: true}, * 1`] = `
Object {
  "cool": true,
}
`;

exports[`gRPC Dynamic Gateway convertBody() should handle {cool: true}, cool 1`] = `true`;

exports[`gRPC Dynamic Gateway convertHeaders() should convert a Authorize token in header to a metadata object 1`] = `
Metadata {
  "_internal_repr": Object {
    "authorize": Array [
      "Bearer DUMMY_TOKEN",
    ],
  },
}
`;

exports[`gRPC Dynamic Gateway convertParams() should handle /v1/hi/{name} 1`] = `
Object {
  "name": "Cool",
  "v1": true,
  "v2": false,
}
`;

exports[`gRPC Dynamic Gateway convertUrl() should correctly convert /{version}/hi/{name}/{cool} into express URL 1`] = `"/:version/hi/:name/:cool"`;

exports[`gRPC Dynamic Gateway convertUrl() should correctly convert /v1/hi/{name} into express URL 1`] = `"/v1/hi/:name"`;

exports[`gRPC Dynamic Gateway getParamsList() should find params in /{version}/hi/{name}/{cool} 1`] = `
Array [
  "version",
  "name",
  "cool",
]
`;

exports[`gRPC Dynamic Gateway getParamsList() should find params in /v1/hi/{name} 1`] = `
Array [
  "name",
]
`;
